api4/team.go

```diff
@@ -1275,7 +1275,18 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	emailList := model.ArrayFromJSON(r.Body)
+	bf, err := io.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.inviteUsersToTeams", "api.team.invite_members_to_team_and_channels.invalid_body.app_error", nil, err.Error(), http.StatusBadRequest)
+		return
+	}
+	memberInvite := &model.MemberInvite{}
+	if jsonErr := json.Unmarshal(bf, &memberInvite); jsonErr != nil {
```

Comment:
what you are suggesting is what it does with the overwrite of the Unmarshal method in the inviteMembers struct. With that Unmarshal function we are able to process both inputs, the array of emails and the memberInvite struct. In case of array of emails, we turn it into a memberInvite struct. If you check the webapp code, the current members invite modal, is still sending the array of emails and it is being parsed correctly, the same as when you open the add members channel and try to invite a member, it will send the memberInvite structure, and both work fine.

<img width="1102" alt="Captura de pantalla 2022-03-29 a las 12 55 17" src="https://user-images.githubusercontent.com/10082627/160674990-29c68c95-0150-4300-aa87-ca0ed1843010.png">
<img width="1073" alt="Captura de pantalla 2022-03-29 a las 12 55 46" src="https://user-images.githubusercontent.com/10082627/160674998-00c913ad-2dfa-4870-b3d0-698a1eb0fe05.png">
<img width="1013" alt="Captura de pantalla 2022-03-29 a las 12 56 00" src="https://user-images.githubusercontent.com/10082627/160675001-4ebf993e-b1ee-4484-9892-0cb59d9837a2.png">